wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -1681,7 +1681,55 @@ public final class MLSService: MLSServiceInterface {
     // MARK: - Proteus to MLS Migration
 
     public func startProteusToMLSMigration() async throws {
-        // Migrate proteus team group conversations to MLS
+        guard let context = context else {
+            assertionFailure("MLSService.context is nil")
+            return
+        }
+
+        let groupConversations = try await context.perform {
+            try ZMConversation.fetchAllTeamGroupConversations(messageProtocol: .proteus, in: context)
+                .map {(
+                    $0.qualifiedID,
+                    $0.mlsGroupID,
+                    $0.localParticipants.map { MLSUser(from: $0) }
+                )}
+        }
+        for (qualifiedID, mlsGroupID, members) in groupConversations {
+
+            guard let qualifiedID else {
+                assertionFailure("the group conversation has no `qualifiedID` set")
+                continue
+            }
+            guard let mlsGroupID else {
+                assertionFailure("the group conversation has no `mlsGroupID` set")
+                continue
+            }
+
+            // update message protocol to `mixed`
+            try await actionsProvider.updateConversationProtocol(qualifiedID: qualifiedID, messageProtocol: .mixed, context: context.notificationContext)
```

Comment:
Should we continue the iterations to try migrate the rest of the groups?

Reply:
I added another do/catch: 4c9005eb66133201980213c1d222b726c98460c1